mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -321,25 +324,25 @@ class PreferencesActivity : AbstractBaseActivity() {
             if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
                 Log.d(TAG, "Removing alarm clock pref")
                 preferenceScreen.removePreference(alarmClockPref)
-            } else {
-                updateAlarmClockPreferenceIcon(alarmClockPref,
-                    alarmClockPref.getPrefValue().toItemUpdatePrefValue())
-                updateAlarmClockPreferenceSummary(alarmClockPref,
-                    sendDeviceInfoPrefixPref.getPrefValue(),
-                    alarmClockPref.getPrefValue().toItemUpdatePrefValue())
-                alarmClockPref.setOnPreferenceChangeListener { preference, newValue ->
-                    val prefix = sendDeviceInfoPrefixPref.getPrefValue()
-                    @Suppress("UNCHECKED_CAST")
-                    val value = newValue as Pair<Boolean, String>
-                    updateAlarmClockPreferenceIcon(preference, newValue)
-                    updateAlarmClockPreferenceSummary(preference, prefix, value)
-                    true
+            }
+
+            phoneStatePref.setOnPreferenceChangeListener { preference, newValue ->
+                @Suppress("UNCHECKED_CAST")
+                val value = newValue as Pair<Boolean, String>
+                if (value.first && ContextCompat.checkSelfPermission(preference.context,
+                        Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
+                    Log.d(TAG, "Request READ_PHONE_STATE permission")
+                    requestPermissions(arrayOf(Manifest.permission.READ_PHONE_STATE),
+                        PERMISSIONS_REQUEST_READ_PHONE_STATE)
                 }
+
+                true
             }
 
             sendDeviceInfoPrefixPref.setOnPreferenceChangeListener { _, newValue ->
-                updateAlarmClockPreferenceSummary(alarmClockPref, newValue as String,
-                    alarmClockPref.getPrefValue().toItemUpdatePrefValue())
+                val prefix = newValue as String
+                (alarmClockPref as ItemUpdatingPreference).updateSummaryAndIcon(prefix)
+                (phoneStatePref as ItemUpdatingPreference).updateSummaryAndIcon(prefix)
```

Comment:
Maybe rather cast the result of `findPreference`? Would be a bit more readable, I guess.

Reply:
I cast the result of `findPreference()` to `Preference` in #1517. After merging that PR, I can rebase this one and adjust the cast.